app/javascript/components/widgets/components/widget/components/widget-header/components/widget-settings/widget-settings-component.jsx

```diff
@@ -17,16 +17,17 @@ class WidgetSettings extends PureComponent {
   }
 
   getUnitVariable = (items, widget, settings, onSettingsChange, type) => {
+    const unit = type === 'categorization system' ? 'source' : type;
     if (items.length <= 1) return null;
     if (items.length === 2) {
       return (
         <Switch
           theme="theme-switch-light"
-          label={type === 'unit' ? 'UNIT' : 'VARIABLE'}
-          value={settings[type]}
+          label={String(type).toUpperCase()}
```

Comment:
Why `String()`?

Reply:
To make sure the type doesn't crash the `toUpperCase`, just a quick decision until we implement the new settings.